tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseEvaluator.scala

```diff
@@ -120,113 +124,105 @@ case object UnlabeledBreadcrumbAccuracy extends ParseStatistic {
-  * makes with respect to gold parses. For instance, say there's a parse whose gold transition
-  * representation is [Sh, Sh, Rt('nsubj), Rt('det)], and a candidate parse comes up with
-  * [Sh, Rt('adjp), Sh, Rt('det)]. The "first" incorrect transition decision is the pair
-  * (Rt('adjp), Sh). In other words, the first mistake that the candidate parse made was to
-  * guess transition Rt('adjp) when it should have guessed Sh.
-  */
-/*
-case object FirstMistakeAggregator extends ParseStatistic {
-  var mistakeHistogram = Map[(StateTransition, StateTransition), Int]()
-
-  override def notify(candidateParse: Option[PolytreeParse], goldParse: PolytreeParse): Unit = {
-    for {
-      candParse <- candidateParse
-      candidateTransitions <- candParse.asTransitionSequence
-      goldTransitions <- goldParse.asTransitionSequence
-      zippedTransitions = candidateTransitions.zip(goldTransitions)
-      firstMistake <- zippedTransitions find { case (t1, t2) => t1 != t2 }
-    } {
-      mistakeHistogram = mistakeHistogram +
-        ((firstMistake._1, firstMistake._2) -> (1 + mistakeHistogram.getOrElse((firstMistake._1,
-          firstMistake._2), 0)))
-    }
-  }
-
-  override def report(): Unit = {}
+/** A ParseScore maps a candidate parse to a score. */
+trait ParseScore extends (PolytreeParse => Double) {
+  def getRatio(candidateParse: PolytreeParse): (Int, Int)
 }
-*/
 
-case object PathAccuracy extends ParseStatistic {
+case class PathAccuracy(ignorePunctuation: Boolean, ignorePathLabels: Boolean)
+    extends ParseStatistic {
+
   var numCorrect = 0
   var numTotal = 0
-  var numCorrectNoPunc = 0
-  var numTotalNoPunc = 0
   var numParses = 0
 
   override def notify(candidateParse: Option[PolytreeParse], goldParse: PolytreeParse): Unit = {
     numParses += 1
-    numTotal += goldParse.breadcrumb.tail.size
-    candidateParse match {
-      case Some(candParse) =>
-        if (candParse.breadcrumb.size == goldParse.breadcrumb.size) {
-          val zipped = candParse.paths.zipWithIndex.tail.zip(
-            goldParse.paths.tail
-          )
-          val nonPuncZipped = zipped filter {
-            case ((x, i), y) =>
-              goldParse.tokens(i).getDeterministicProperty('cpos) != Symbol(".")
-          }
-          // skip the first element because it is the nexus (hence it has no breadcrumb)
-          numCorrect += zipped count
-            { case ((x, _), y) => (x == y) }
-          numCorrectNoPunc += nonPuncZipped count
-            { case ((x, _), y) => (x == y) }
-          numTotalNoPunc += nonPuncZipped.size
-        } else { // skip the parse if the tokenization is different
-          println(s"WARNING -- Skipping parse: ${candParse.sentence.asWhitespaceSeparatedString}" +
-            s" tokenized differently than gold: ${goldParse.sentence.asWhitespaceSeparatedString}")
-          numParses -= 1
-          numTotal -= goldParse.breadcrumb.tail.size
-        }
-      case None =>
+    if (candidateParse != None) {
```

Comment:
I think we like pattern matching in this case.
```
candidateParse match {
   Some(parse) => ...
   None => /* nothing */
}
```
I don't feel super strongly about it, but I guess a good argument for this is that it shows you've thought about the `None` case, and it's not just omitted by accident.